Avoid chunk argument

bjorne opened the pull request:
The argument to Enumerable#chunk was removed in 2.3:

> - enum.c: Enumerable#chunk and Enumerable#slice_before no longer takes
>   the initial_state argument. [Feature #10958]

This also adds 2.3.1 to the Travis build matrix.

bjorne committed:
Avoid usage of chunk state argument

The argument to Enumerable#chunk was removed in 2.3:

> * enum.c: Enumerable#chunk and Enumerable#slice_before no longer takes
>   the initial_state argument. [Feature #10958]

bjorne committed:
Add 2.3.1 to travis build matrix

mthssdrbrg commented on lib/tina/restore_plan.rb:
```diff
@@ -46,13 +46,15 @@ def size
-              state[:sum] = object.size
-              state[:index] += 1
+          sum = 0
+          index = 0
```
`sum = index = 0` to save a line? :)

bjorne replied:
I prefer the current format. #wontfix

Fonsan commented on lib/tina/restore_plan.rb:
```diff
@@ -46,13 +46,15 @@ def size
 
       def chunk(max_chunk_size)
         @chunks ||= begin
```
Memoizing the result without respecting argument is a accident waiting to happen

bjorne replied:
Nice catch.

Fonsan replied:
Also the memoizing instance variable should have the same name as the method

bjorne replied:
I removed the memoization altogether.

bjorne committed:
Update bundler on travis

See:
* bundler/bundler#3559
* rubygems/rubygems#1419
* travis-ci/travis-ci#3531

bjorne committed:
Remove memoization that does not respect argument

Should be fine since we in practive only invoke #chunk once.

bjorne committed:
Fix inconsistent chunking

The chunking performed within RestorePlan#price and Cli#restore was
inconsistent. This did not show because of the memoization that existed
previously.
